zipline/data/us_equity_pricing.py

```diff
@@ -978,12 +978,21 @@ def calc_dividend_ratios(self, dividends):
         equity_daily_bar_reader = self._equity_daily_bar_reader
 
         effective_dates = full(len(amounts), -1, dtype=int64)
+
         calendar = self._calendar
+
+        # Calculate locs against a tz-naive cal, as the ex_dates are tz-
+        # naive.
+        tz_naive_calendar = calendar.tz_localize(None)
```

Comment:
Rather than localizing the calendar to naive, can we localize the ex_dates to the calendar's time zone?  They should be semantically identical, but using the same calendar prevents allocation of a new hash table.